src/tools/rbd/action/Trash.cc

```diff
@@ -320,6 +345,148 @@ int execute_list(const po::variables_map &vm) {
   return 0;
 }
 
+void get_purge_arguments(po::options_description *positional,
+                            po::options_description *options) {
+  at::add_pool_options(positional, options);
+  at::add_no_progress_option(options);
+  
+  options->add_options()
+      (EXPIRED_BEFORE.c_str(), po::value<std::string>()->value_name("date"), 
+       "purges images that expired before the given date");
+  options->add_options()
+      (THRESHOLD.c_str(), po::value<double>(), 
+       "purges images until the current pool data usage is reduced to X%, "
+       "value range: 0.0-1.0");
+}
+
+
+int execute_purge (const po::variables_map &vm) {
+  size_t arg_index = 0;
+  std::string pool_name = utils::get_pool_name(vm, &arg_index);
+
+  librados::Rados rados;
+  librados::IoCtx io_ctx;
+  int r = utils::init(pool_name, &rados, &io_ctx);
+  if (r < 0) {
+    return r;
+  }
+
+  librbd::RBD rbd;
+  
+  std::vector<librbd::trash_image_info_t> trash_entries;
+  r = rbd.trash_list(io_ctx, trash_entries);
+  if (r < 0) { 
+     return r;
+  }
+
+  std::vector<const char *> to_be_removed;
+
+  if (vm.find(THRESHOLD) != vm.end()) {
+    double threshold = vm[THRESHOLD].as<double>();
+    if (threshold < 0 || threshold > 1) {
+        std::cout << "rbd: argument 'threshold' is out of valid range"
+                  << std::endl;
+        return -EINVAL;
+    }
+
+    librados::bufferlist inbl;
+    librados::bufferlist outbl;
+    rados.mon_command("{\"prefix\": \"df\", \"format\": \"json\"}", inbl,
+                      &outbl, NULL);
+   
+
+    json_spirit::mValue json;
+    json_spirit::read_or_throw(outbl.to_str(), json);
```

Comment:
Nit: would be nicer to not throw and instead log an error and return an error code if you couldn't parse the result